### ISSUE TITLE
'flutter run -t' is not working as expected

### ISSUE THREAD

**Issue by jstorch72** [flutter/flutter#93550]
**flutter does not launch correctly when using the "-t" option unless the file is named "lib/main.dart"**

_Current behavior:_
Launching lib\prod.dart on AOSP on IA Emulator in debug mode...
Running Gradle task 'assembleDebug'...
.dart_tool/flutter_build/generated_main.dart:8:8: Error: Error when reading 'lib/main.dart': The system cannot find the file specified.

import 'package:fluttermobile/main.dart' as entrypoint;
       ^
.dart_tool/flutter_build/generated_main.dart:50:18: Error: Getter not found: 'main'.
  if (entrypoint.main is _UnaryFunction) {

_Expected behavior:_
Application to launch normally.

Flutter doctor -v info:
√] Flutter (Channel stable, 2.5.0, on Microsoft Windows [Version 10.0.19041.1348], locale en-US)
    • Flutter version 2.5.0 at C:\frameworks\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4cc385b4b8 (9 weeks ago), 2021-09-07 23:01:49 -0700
    • Engine revision f0826da7ef
    • Dart version 2.14.0

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at D:\frameworks\andriod\sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = D:\frameworks\andriod\sdk
    • ANDROID_SDK_ROOT = D:\frameworks\andriod\emmulators
    • Java binary at: C:\Program Files\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code, 64-bit edition (version 1.61.2)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.27.0

[√] Connected device (4 available)
    • SM G900V (mobile)            • 470196df      • android-arm    • Android 6.0.1 (API 23)
    • AOSP on IA Emulator (mobile) • emulator-5554 • android-x86    • Android 9 (API 28) (emulator)
    • Chrome (web)                 • chrome        • web-javascript • Google Chrome 95.0.4638.69
    • Edge (web)                   • edge          • web-javascript • Microsoft Edge 95.0.1020.44

• No issues found!

Please let me know if I need to supply additional information.

**Comment by stuartmorgan** [comment]
Duplicate of https://github.com/flutter/flutter/issues/90920

The workaround listed there should unblock you until the fix reaches the `stable` channel.
